tox_docker/tox4/plugin.py

```diff
@@ -0,0 +1,122 @@
+from typing import Dict, List
+
+from tox.config.cli.parser import ToxParser
+from tox.config.main import Config
+from tox.config.sets import ConfigSet
+from tox.config.types import EnvList
+from tox.execute.api import Outcome
+from tox.plugin import impl
+from tox.tox_env.api import ToxEnv
+
+from tox_docker.config import ContainerConfig, RunningContainers
+from tox_docker.plugin import (
+    docker_health_check,
+    docker_pull,
+    docker_run,
+    get_env_vars,
+    HealthCheckFailed,
+    stop_containers,
+)
+from tox_docker.tox4.config import (
+    discover_container_configs,
+    EnvRunningContainers,
+    parse_container_config,
+)
+from tox_docker.tox4.log import log
+
+CONTAINER_CONFIGS: Dict[str, ContainerConfig] = {}
+ENV_CONTAINERS: EnvRunningContainers = {}
+
+
+@impl
+def tox_add_core_config(core_conf: ConfigSet, config: Config) -> None:
+    container_config_names = discover_container_configs(config)
```

Comment:
This section here should only handle core configuration. You're registering environment configuration, you have a dedicated config for env configuration. Please don't do this here.